polynote-frontend/polynote/ui/component/notebook/comment.ts

```diff
@@ -279,6 +281,10 @@ class CommentRoot extends MonacoRightGutterOverlay {
         handledChangedComments(this.allCommentsState.state)
         allCommentsState.addObserver((curr, updateResult) => handledChangedComments(curr, updateResult))
 
+        this.rootState.addObserver((curr: CellComment, updateResult: UpdateResult<CellComment>) => {
```

Comment:
I'd bump this up right next to where you set `this.range` initially, just for clarity.